pkg/controller/credentialsrequest/credentialsrequest_controller_test.go

```diff
@@ -684,6 +685,24 @@ func TestCredentialsRequestReconcile(t *testing.T) {
 				assert.NoError(t, err)
 			},
 		},
+		{
+			name: "skip sync if last sync less than 1 hour ago (and lastsyncgeneration is current)",
+			existing: []runtime.Object{
+				createTestNamespace(testNamespace),
+				createTestNamespace(testSecretNamespace),
+				testCredentialsRequestWithRecentLastSync(t),
+			},
+			mockRootAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				return mockAWSClient
+			},
+			validate: func(c client.Client, t *testing.T) {
+				cr := getCR(c)
+				// the credentialrequest was last synced 20 minutes ago so pushing the
+				// time forward 5 minutes should still leave us in the past (ie no sync happened)
+				assert.True(t, cr.Status.LastSyncTimestamp.Add(5*time.Minute).Before(time.Now()), "expect lastsynctimestamp to be in the past")
```

Comment:
LastSyncTimestamp should be unmodified if we skipped, so I think you just want to test that it's equal to twentyMinuteOldTimestamp. The empty mock on AWS will validate we didn't try to hit AWS nicely.